### ISSUE TITLE
php70-imagick has warnings

### ISSUE THREAD

**Issue by ChuckHeintzelman** [Homebrew/homebrew-php#2766]
After latest brew update I'm getting the following error from every php script that executes:

Warning: Version warning: Imagick was compiled against Image Magick version 1682 but version 1683 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0

**Comment by Riimu** [comment]
I had similar issue with php56-imagick. I assume the bottle just hasn't been updated yet to reflect the latest imagemagick version. I fixed the issue on my installation by building it from source myself by running:

    brew rm php56-imagick
    brew install --build-from-source php56-imagick

**Comment by ChuckHeintzelman** [comment]
Thank you @Riimu 

I did the following:

~~~
brew rm php70-imagick
brew install --build-from-source php70-imagick
~~~

And now the warning is gone!

**Comment by james2doyle** [comment]
Had this same problem with PHP-70. Fixed it with the following:

`brew rm php70-imagick && brew install --build-from-source php70-imagick`

Now everything is quiet again!

**Comment by javian** [comment]
The bottles have now been updated so it should no longer be an issue.

**Comment by cdekok** [comment]
@javian i still have this issue today

**Comment by javian** [comment]
Reopening for further investigation

**Comment by prawee** [comment]
@ChuckHeintzelman thank you that's working for me.

**Comment by javian** [comment]
Is this still an issue ?

**Comment by cdekok** [comment]
Seems to be solved for me ater brew update / upgrade and installing php70-imagick again.

**Comment by jajajaime** [comment]
Happening to me still:
PHP Warning:  Version warning: Imagick was compiled against Image Magick version 1685 but version 1686 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0

**Comment by daveaspinall** [comment]
Happened to me again this morning too, after a `brew update && brew upgrade`.

```
PHP Warning:  Version warning: Imagick was compiled against Image Magick version 1685 but version 1686 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0
```

Running `brew rm php70-imagick && brew install --build-from-source php70-imagick` seems to have fixed it though.

**Comment by samuelg0rd0n** [comment]
Happened to me today as well and installing from the source fixed it. Thank you for the solution.

**Comment by bytehead** [comment]
Same on php71: 
`PHP Warning:  Version warning: Imagick was compiled against Image Magick version 1796 but version 1797 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0`

This fixed it:
`brew uninstall --force php71-imagick`
`brew install  --build-from-source php71-imagick`

**Comment by alcaeus** [comment]
Alternatively: `brew reinstall -s php71-imagick`. #4014 fixes this (again).

**Comment by Preclowski** [comment]
Actually any of solutions doesn't work :(

**Comment by indapublic** [comment]
```
brew rm php71-imagick
brew cleanup
brew install --build-from-source php71-imagick
brew services restart php71
brew services restart nginx
```

And all is fixed. Thanks to all above

**Comment by james2doyle** [comment]
Yeah, this is still happening. I get it every time `imagick` is updated.

**Comment by alcaeus** [comment]
See #4225. FWIW, you don't need to uninstall and clean as @indapublic suggested, a simple `brew reinstall -s php<your-version-here>-imagick` does the trick.

**Comment by Preclowski** [comment]
As I said, it doesn't work. Even with cleanup.

> PHP Warning:  Version warning: Imagick was compiled against Image Magick version 1798 but version 1578241 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0

**Comment by vesper8** [comment]
@Preclowski `brew reinstall -s php71-imagick` solved it at once for me.. surprised it's not working for you

**Comment by beavis** [comment]
I'm having the same issue:
> PHP Warning:  Version warning: Imagick was compiled against Image Magick version 1798 but version 1644032 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0
Reinstalling from source won't fix it, notice the odd loaded version number
Any idea for a fix?

**Comment by beavis** [comment]
@Preclowski do you have graphicsmagick installed by any chance? (seems like that's my problem)

**Comment by Preclowski** [comment]
Yes, I have graphicsmagic installed. Also php71-gmagick

**Comment by beavis** [comment]
@Preclowski @javian I figured that given the odd version number on the warning it could be a problem with graphicsmagick, once you uninstall graphicsmagick that version warning disappears so there must be some kind of conflict, I have had both installed for a while but the warnings started recently if I recall correctly

**Comment by ptheofan** [comment]
Also facing same issue. Older version used to work - Some months ago I installed php71 and had that issue. Didn't care much, was just playing with it so I ignored it. However things change and I now actually need it so I tried to fix it. I ended up removing it from php70 also (and earlier versions). After reinstalling even in php70 (which used to work fine) I now get the same error.

```
PHP Warning:  Version warning: Imagick was compiled against Image Magick version 1799 but version 2103040 is loaded. Imagick will run but may behave surprisingly in Unknown on line 0
```
